packages/test/test-end-to-end-tests/src/test/gc/gcDataVirtualization.spec.ts

```diff
@@ -0,0 +1,219 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+	createTestConfigProvider,
+	createSummarizer,
+	ITestContainerConfig,
+	ITestObjectProvider,
+	summarizeNow,
+	waitForContainerConnection,
+} from "@fluidframework/test-utils";
+import {
+	describeCompat,
+	ITestDataObject,
+	TestDataObjectType,
+} from "@fluid-private/test-version-utils";
+import {
+	ISummaryTree,
+	SummaryType,
+	type ISnapshotTree,
+} from "@fluidframework/protocol-definitions";
+import { IContainer, LoaderHeader } from "@fluidframework/container-definitions";
+import type { ISnapshot } from "@fluidframework/driver-definitions";
+import { getGCStateFromSummary } from "./gcTestSummaryUtils.js";
+
+const interceptResult = <T>(
+	parent: any,
+	fn: (...args: any[]) => Promise<T>,
+	intercept: (result: T) => void,
+) => {
+	const interceptFn = async (...args: any[]) => {
+		const val = await fn.apply(parent, args);
+		intercept(val);
+		return val as T;
+	};
+	parent[fn.name] = interceptFn;
+	interceptFn.bind(parent);
+	return fn;
+};
+
+const assertOmittedTree = (
+	snapshotTree: ISnapshotTree,
+	groupId: string | undefined,
+	message: string,
+) => {
+	assert(snapshotTree.omitted, message);
+	assert(snapshotTree.groupId === groupId, message);
+	assert(Object.entries(snapshotTree.trees).length === 0, message);
+	assert(Object.entries(snapshotTree.blobs).length === 0, message);
+};
+
+/**
+ * Validates that an unreferenced datastore goes through all the GC phases without overlapping.
+ */
+describeCompat("GC & Data Virtualization", "NoCompat", (getTestObjectProvider) => {
+	const configProvider = createTestConfigProvider();
+	configProvider.set("Fluid.Container.UseLoadingGroupIdForSnapshotFetch", true);
+	const testContainerConfig: ITestContainerConfig = {
+		runtimeOptions: {
+			summaryOptions: {
+				summaryConfigOverrides: {
+					state: "disabled",
+				},
+			},
+		},
+		loaderProps: { configProvider },
+	};
+
+	let provider: ITestObjectProvider;
+
+	const loadSummarizer = async (container: IContainer, summaryVersion?: string) => {
+		return createSummarizer(
+			provider,
+			container,
+			{
+				loaderProps: { configProvider },
+			},
+			summaryVersion,
+		);
+	};
+
+	function getDataStoreInSummaryTree(summaryTree: ISummaryTree, dataStoreId: string) {
+		const channelsTree = summaryTree.tree[".channels"];
+		assert(channelsTree !== undefined, "Expected a .channels tree");
+		assert(channelsTree.type === SummaryType.Tree, "Expected a tree");
+		return channelsTree.tree[dataStoreId];
+	}
+
+	async function isDataStoreInSummaryTree(summaryTree: ISummaryTree, dataStoreId: string) {
+		return getDataStoreInSummaryTree(summaryTree, dataStoreId) !== undefined;
+	}
+
+	beforeEach("setup", async function () {
+		provider = getTestObjectProvider({ syncSummarizer: true });
+
+		// Data virtualization only works with local
+		// TODO: enable for ODSP
```

Comment:
Is there an ADO tracking this? With the risk of nitpicking, this should probably call out the ADO# here or remove the TODO altogether.

Reply:
Updated